src/base/static/components/input-form/index.js

```diff
@@ -314,73 +304,90 @@ class InputForm extends Component {
-          this.selectedCategoryConfig.datasetSlug,
+      // Save attachments.
+      if (this.attachments.length) {
+        const attachmentPromises = await mapseedApiClient.attachments.create(
+          response.url,
+          this.attachments,
         );
-        this.props.createPlace(response.toJSON());
-
-        // Save autofill values as necessary.
-        // TODO: This logic is better suited for the FormField component,
-        // perhaps in an onSave hook.
-        this.selectedCategoryConfig.fields.forEach(fieldConfig => {
-          if (fieldConfig.autocomplete) {
-            Util.saveAutocompleteValue(
-              fieldConfig.name,
-              this.state.fields
-                .get(fieldConfig.name)
-                .get(constants.FIELD_VALUE_KEY),
-              constants.AUTOFILL_DURATION_DAYS,
-            );
-          }
-        });
 
-        this.setState({ isFormSubmitting: false, showValidityStatus: false });
+        response.attachments = await Promise.all(attachmentPromises);
+      }
+
+      // Only add this place to the places duck if it isn't private.
+      !response.private && this.props.createPlace(response);
+
+      this.setState({ isFormSubmitting: false, showValidityStatus: false });
 
-        // Fire post-save hook.
-        // The post-save hook allows flavors to hijack the default
-        // route-to-detail-view behavior.
-        if (this.props.customHooks && this.props.customHooks.postSave) {
-          this.props.customHooks.postSave(
-            response,
-            model,
-            this.defaultPostSave.bind(this),
+      emitter.emit(
+        constants.PLACE_COLLECTION_ADD_PLACE_EVENT,
+        this.selectedCategoryConfig.datasetSlug,
+      );
+
+      // Save autofill values as necessary.
+      // TODO: This logic is better suited for the FormField component,
+      // perhaps in an onSave hook.
+      this.selectedCategoryConfig.fields.forEach(fieldConfig => {
+        if (fieldConfig.autocomplete) {
+          Util.saveAutocompleteValue(
+            fieldConfig.name,
+            this.state.fields
+              .get(fieldConfig.name)
+              .get(constants.FIELD_VALUE_KEY),
+            constants.AUTOFILL_DURATION_DAYS,
           );
-        } else {
-          this.defaultPostSave(model);
         }
-      },
-      error: () => {
-        Util.log("USER", "new-place", "fail-to-add-place");
-      },
-      wait: true,
-    });
-  }
+      });
+
+      // Fire post-save hook.
+      // The post-save hook allows flavors to hijack the default
+      // route-to-detail-view behavior.
+      if (this.props.customHooks && this.props.customHooks.postSave) {
+        this.props.customHooks.postSave(
+          response,
+          response,
+          this.defaultPostSave.bind(this),
+        );
+      } else {
+        this.defaultPostSave(response);
+      }
+    } else {
+      alert("Oh dear. It looks like that didn't save. Please try again.");
```

Comment:
All of the user feedback on API client failure is super cheesy in this PR, but it felt like too much to also think through how to handle errors in a graceful way, so I decided to punt this for now.